File: common/src/main/java/juuxel/adorn/block/FluidUtil.java
package juuxel.adorn.block;

import net.minecraft.world.level.block.state.BlockState;
import net.minecraft.world.level.material.FluidState;
import net.minecraft.world.level.material.Fluids;
import net.minecraft.world.level.block.state.properties.BlockStateProperties;

public final class FluidUtil {
    /**
     * For using a mixin (juuxel.adorn.mixin.fluidloggable.FluidUtilMixin) to set the fluid property
     * for Towelette support.
     */
    public static BlockState updateFluidFromState(BlockState state, FluidState fluidState) {
        return state.setValue(BlockStateProperties.WATERLOGGED, fluidState.getType() == Fluids.WATER);
    }
}

File: common/src/main/kotlin/juuxel/adorn/block/FluidUtil.kt
package juuxel.adorn.block

import net.minecraft.block.BlockState
import net.minecraft.fluid.FluidState
import net.minecraft.fluid.Fluids
import net.minecraft.state.property.Properties

object FluidUtil {
    /**
     * For using a mixin ([juuxel.adorn.mixin.fluidloggable.FluidUtilMixin]) to set the fluid property
     * for Towelette support.
     */
    @JvmStatic
    fun updateFluidFromState(state: BlockState, fluidState: FluidState) =
        state.with(Properties.WATERLOGGED, fluidState.fluid == Fluids.WATER)
}
